File: c/exemples/exemple_arbre_binaire.c
#include "arbre_binaire.h"

#include <stdio.h>

int main(void) {
    ArbreBinaire arbre = creer_arbre_binaire();

    definir_racine_arbre_binaire(&arbre, 8);
    ajouter_fils_gauche_arbre_binaire(arbre.racine, 3);
    ajouter_fils_droit_arbre_binaire(arbre.racine, 10);
    ajouter_fils_gauche_arbre_binaire(arbre.racine->gauche, 1);
    ajouter_fils_droit_arbre_binaire(arbre.racine->gauche, 6);

    printf("Parcours prefixe : ");
    parcours_prefixe_arbre_binaire(&arbre);
    printf("Parcours infixe : ");
    parcours_infixe_arbre_binaire(&arbre);
    printf("Parcours suffixe : ");
    parcours_suffixe_arbre_binaire(&arbre);
    printf("Taille de l'arbre : %zu\n", taille_arbre_binaire(&arbre));

    detruire_arbre_binaire(&arbre);
    return 0;
}

File: ocaml/exemples/exemple_arbre_binaire.ml
let chaine_de_liste_entiere valeurs =
  "[" ^ String.concat " ; " (List.map string_of_int valeurs) ^ "]"

let () =
  let arbre =
    Arbre_binaire.creer_noeud_arbre_binaire
      8
      (Arbre_binaire.creer_noeud_arbre_binaire
         3
         (Arbre_binaire.creer_noeud_arbre_binaire 1 Arbre_binaire.Vide Arbre_binaire.Vide)
         (Arbre_binaire.creer_noeud_arbre_binaire 6 Arbre_binaire.Vide Arbre_binaire.Vide))
      (Arbre_binaire.creer_noeud_arbre_binaire 10 Arbre_binaire.Vide Arbre_binaire.Vide)
  in
  Printf.printf "Parcours prefixe : %s\n"
    (chaine_de_liste_entiere (Arbre_binaire.parcours_prefixe_arbre_binaire arbre));
  Printf.printf "Parcours infixe : %s\n"
    (chaine_de_liste_entiere (Arbre_binaire.parcours_infixe_arbre_binaire arbre));
  Printf.printf "Parcours suffixe : %s\n"
    (chaine_de_liste_entiere (Arbre_binaire.parcours_suffixe_arbre_binaire arbre));
  Printf.printf "Taille de l'arbre : %d\n" (Arbre_binaire.taille_arbre_binaire arbre)
